File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/Post_blog_comment_021_Secure.py
import re


comments = []


EMAIL_PATTERN = r"^[\w\.-]+@[\w\.-]+\.\w+$"


def post_blog_comment(name, email, comment_text):
    """
    Securely posts blog comments.
    """

    try:

        if name is None or email is None or comment_text is None:
            return "Required fields are missing."

        name = name.strip()
        email = email.strip()
        comment_text = comment_text.strip()

        if not name or not email or not comment_text:
            return "Required fields are missing."

        if not re.match(EMAIL_PATTERN, email):
            return "Invalid email address."

        comments.append({
            "name": name,
            "email": email,
            "comment": comment_text
        })

        return "Comment posted successfully."

    except Exception:
        return "Unable to post comment."


if __name__ == "__main__":
    print(post_blog_comment(
        "Ali",
        "ali@example.com",
        "Excellent post!"
    ))

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Post_blog_comment_021_Secure.java
import java.util.ArrayList;
import java.util.List;
import java.util.regex.Pattern;

public class Post_blog_comment_021_Secure {

    static class Comment {
        String name;
        String email;
        String comment;

        Comment(String name, String email, String comment) {
            this.name = name;
            this.email = email;
            this.comment = comment;
        }
    }

    static List<Comment> database = new ArrayList<>();

    static final Pattern EMAIL =
            Pattern.compile("^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$");

    public static String postComment(String name, String email, String comment) {

        if (name == null || email == null || comment == null)
            return "Required fields are missing.";

        name = name.trim();
        email = email.trim();
        comment = comment.trim();

        if (name.isEmpty() || email.isEmpty() || comment.isEmpty())
            return "Required fields are missing.";

        if (!EMAIL.matcher(email).matches())
            return "Invalid email address.";

        database.add(new Comment(name, email, comment));

        return "Comment posted successfully.";
    }

    public static void main(String[] args) {

        System.out.println(postComment(
                "Ali",
                "ali@test.com",
                "Excellent article!"
        ));
    }
}